### PR TITLE
Refactored reducers and added attending checkbox functionality

### PR DESCRIPTION
**Pull request by lxwyndxl**
@jessicatai

### CODE REVIEW

File: src/app/actions/modal.js [hunk 1]
```diff
@@ -1,3 +1,11 @@
+export const hideModal = () => {
```
Comment (jessicatai) [inline src/app/actions/modal.js]: i'm a big fan of this

File: src/app/components/modals/rsvp/rsvp-attendees.js [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
+import { updateAttendingStatus,
+         toggleEmailEditState } from '../../../actions/rsvp-attendees';
+
 import Avatar from 'material-ui/Avatar';
 import Checkbox from 'material-ui/Checkbox';
 
-export function Attendee({ name, email, isAttending, onCheck }) {
+export function Attendee({ name, email, isAttending, onCheck, id }) {
   const initial = name[0].toUpperCase();
+  const checkboxName = `user-${id}`;
```
Comment (jessicatai) [inline src/app/components/modals/rsvp/rsvp-attendees.js]: can we use regular id? then you don't need to parse it in `onCheck`

File: src/app/components/modals/rsvp/rsvp-content.js [hunk 3]
```diff
@@ -26,6 +26,7 @@ class RsvpContent extends Component {
           />
           <RsvpDetails
             userGroup={userGroup}
+            users={users}
```
Comment (jessicatai) [inline src/app/components/modals/rsvp/rsvp-content.js]: `You're` should be `You’re`

File: src/app/components/modals/rsvp/rsvp-details.js [hunk 4]
```diff
@@ -17,9 +17,11 @@ class RsvpDetails extends Component {
       notes,
     } = this.props.userGroup;
 
+    const users = this.props.users;
```
Comment (jessicatai) [inline src/app/components/modals/rsvp/rsvp-details.js]: nit: const { users }  = this.props;

File: src/app/reducers/users.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+const initialState = [];
+
+const users = (state = initialState, action) => {
+  switch (action.type) {
+    case 'RECEIVE_RSVP_GROUP':
```
Comment (jessicatai) [inline src/app/reducers/users.js]: constants for these? looks like user-group.js is also using this string

File: src/app/components/modals/rsvp/rsvp-attendees.js (hunk 2 above)
Comment (lxwyndxl) [inline src/app/components/modals/rsvp/rsvp-attendees.js]: id might be not unique enough, which might cause clashing

**Pull request closed without merging**